src/background/in-dapp-notifications.ts

```diff
@@ -0,0 +1,65 @@
+import browser from 'webextension-polyfill';
+import type { ChainId } from 'src/modules/ethereum/transactions/ChainId';
+import type { InDappNotification } from 'src/shared/types/InDappNotification';
+import type { Wallet } from 'src/shared/types/Wallet';
+import { getNetworksStore } from 'src/modules/networks/networks-store.background';
+import type { Chain } from 'src/modules/networks/Chain';
+import { emitter } from './events';
+import { INTERNAL_SYMBOL_CONTEXT } from './Wallet/Wallet';
+
+export class InDappNotificationService {
+  private readonly getWallet: () => Wallet;
+
+  constructor({ getWallet }: { getWallet: () => Wallet }) {
+    this.getWallet = getWallet;
+  }
+
+  async getTabIdsByOrigin(origin: string) {
+    const tabs = await browser.tabs.query({});
+    return tabs
+      .filter((tab): tab is browser.Tabs.Tab & { id: number } =>
+        Boolean(tab.url && tab.id && new URL(tab.url).origin === origin)
+      )
+      .map((tab) => tab.id);
+  }
+
+  async notify(origin: string, notification: InDappNotification) {
+    const tabIds = await this.getTabIdsByOrigin(origin);
+    tabIds.map((tabId) => chrome.tabs.sendMessage(tabId, notification));
```

Comment:
This is a side-effect, so:
```suggestion
    tabIds.forEach((tabId) => chrome.tabs.sendMessage(tabId, notification));
```

Or even a `for of` loop can be used

Reply:
✅